lib/qiita/markdown/summary_processor.rb

```diff
@@ -13,6 +20,26 @@ class SummaryProcessor < Processor
         Filters::Truncate
       ]
 
+      # @param [Hash] context Optional context for HTML::Pipeline.
+      def initialize(context = {})
+        @context = DEFAULT_CONTEXT.merge(context)
+      end
+
+      # Converts Markdown text into HTML string with extracted metadata.
+      #
+      # @param [String] input Markdown text.
+      # @param [Hash] context Optional context merged into default context.
+      # @return [Hash] Process result.
+      # @example
+      #   Qiita::Markdown::SummaryProcessor.new.call(markdown) #=> {
+      #     codes: [...],
+      #     mentioned_usernames: [...],
+      #     output: "...",
+      #   }
+      def call(input, context = {})
+        HTML::Pipeline.new(filters, @context).call(input, context)
+      end
+
```

Comment:
Now there's no reason to let `SummaryProcessor` inherit `Processor` since `SummaryProcessor` implements all the methods of `Processor`.